### PR TITLE
fix: update React Native SDK for `clientlib` 0.2.0

### PR DESCRIPTION
**Pull request by steveluscher**

### CODE REVIEW

**Comment by steveluscher** [comment]
Can you give this a once-over @sdlaver? I sort of just crashed straight through this refactor without trying to understand the changes to `clientlib` too deeply.

File: js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt [hunk 1]
```diff
@@ -47,42 +46,37 @@ class SolanaMobileWalletAdapterModule(reactContext: ReactApplicationContext) :
...
             cleanup()
             promise.reject("Timed out waiting for local association to be ready", e)
+        } catch (e: InterruptedException) {
+            Log.w(name, "Interrupted while waiting for local association to be ready", e)
+            cleanup()
+            promise.reject(e)
+        } catch (e: ExecutionException) {
+            Log.e(name, "Failed establishing local association with wallet", e.cause)
+            cleanup()
+            promise.reject(e)
```
Comment (steveluscher) [inline js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt:79]: Are these different or should one of them be eliminated now?

File: js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt [hunk 2]
```diff
@@ -47,42 +46,37 @@ class SolanaMobileWalletAdapterModule(reactContext: ReactApplicationContext) :
...
+            val client =
+                localAssociation.start().get(ASSOCIATION_TIMEOUT_MS.toLong(), TimeUnit.MILLISECONDS)
+            sessionState = SessionState(client, localAssociation)
             promise.resolve(true)
         } catch (e: ActivityNotFoundException) {
             Log.e(name, "Found no installed wallet that supports the mobile wallet protocol", e)
             cleanup()
             promise.reject("ERROR_WALLET_NOT_FOUND", e)
-        } catch (e: TimeoutCancellationException) {
+        } catch (e: TimeoutException) {
```
Comment (steveluscher) [inline js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt:68]: Presumably this is the new exception to watch now that the `get()` call is blocking?

Comment (sdlaver) [reply js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt:68]: Yep. `java.util.concurrent.TimeoutException` from `get()`, vs the old `kotlinx.coroutines.TimeoutCancellationException` from `withTimeout`.

File: js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt (hunk 1 above)
Comment (sdlaver) [reply js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt:79]: These can still both be thrown.

`InterruptedException` is a generic Java exception for when `Thread.interrupt` is called, to cancel a normally blocking I/O method.

`ExecutionException` is the ugly exception output from `Future.get`; it wraps all exceptions other than the ones explicitly defined for `Future.get`. It makes sane exception handling with Futures harder than it needs to be :(

**Pull request merged**